packages/nuxt/src/core/nitro.ts

```diff
@@ -288,16 +288,18 @@ export async function initNitro (nuxt: Nuxt & { _nitro?: Nitro }) {
           createRadixRouter({ routes: routeRules })
         )
         const payloadSuffix = nuxt.options.experimental.renderJsonPayloads ? '/_payload.json' : '/_payload.js'
-        for (const route of nitro._prerenderedRoutes || []) {
-          if (!route.error && route.route.endsWith(payloadSuffix)) {
-            const url = route.route.slice(0, -payloadSuffix.length) || '/'
-            const rules = defu({}, ...routeRulesMatcher.matchAll(url).reverse()) as Record<string, any>
-            if (!rules.prerender) {
-              prerenderedRoutes.add(url)
+        if (nitro._prerenderedRoutes?.length) {
```

Comment:
I don't see an issue with the previous code here...

Reply:
This is kind of the same principle as everything else, if the array is empty the performance overhead to begin checking the iteration (even if exiting immediately) is much bigger than checking for a truthy length. I'll be able to append benchmark results when I get on PC but there was a notable increase in performance.

Reply:
Update: here it is.

```ts
const routeRulesMatcher=""
function defu(table, ...args){

}
const prerenderedRoutes=new Set()
const payloadSuffix=""
const nitro={
    _prerenderedRoutes:[]
}
console.time("current")
for (let index = 0; index < 100000; index++) {

    for (const route of nitro._prerenderedRoutes || []) {
        if (!route.error && route.route.endsWith(payloadSuffix)) {
          const url = route.route.slice(0, -payloadSuffix.length) || '/'
          const rules = defu({}, ...routeRulesMatcher.matchAll(url).reverse())
          if (!rules.prerender) {
            prerenderedRoutes.add(url)
          }
        }
    }
}
console.timeEnd("current")
console.time("PR")
for (let index = 0; index < 100000; index++) {
    if (nitro._prerenderedRoutes?.length) {
        for (const route of nitro._prerenderedRoutes) {
          if (!route.error && route.route.endsWith(payloadSuffix)) {
            const url = route.route.slice(0, -payloadSuffix.length) || '/'
            const rules = defu({}, ...routeRulesMatcher.matchAll(url).reverse())
            if (!rules.prerender) {
              prerenderedRoutes.add(url)
            }
          }
        }
    }
}

console.timeEnd("PR")
```

![image](https://github.com/nuxt/nuxt/assets/92037085/5174698a-8168-4c16-affc-21755a757d98)